C/plugins/storage/sqlite/common/readings_catalogue.cpp

```diff
@@ -1874,7 +1875,16 @@ ReadingsCatalogue::tyReadingReference  ReadingsCatalogue::getReadingReference(Co
 
 						startReadingsId = 1;
 
-						success = createNewDB(dbHandle,  dbId, startReadingsId, NEW_DB_ATTACH_REQUEST);
+						if (getEmptyReadingTableReference(emptyTableReference))
+						{
+							emptyTableReference.tableId;
```

Comment:
I don't want to assign anything here because I am passing variable emptyTableReference by reference to function getEmptyReadingTableReference so I will get back updated values if any. To avoid confusion I changed if-else construct to if